src/autogluon/bench/eval/scripts/run_evaluation_openml.py

```diff
@@ -15,20 +15,115 @@
+    ),
+    compute_z_score: bool = typer.Option(
+        True,
+        help="Whether to compute comparison z-scores. The framework being compared against is the first one listed in `frameworks_run`. Only valid when multiple folds are used and `treat_folds_as_datasets=False`.",
+    ),
+    treat_folds_as_datasets: bool = typer.Option(False, help="If True, treat each fold as a separate dataset."),
+    use_tid_as_dataset_name: bool = typer.Option(
+        True,
+        help="If True, replaces dataset human-readable names with unique integer IDs associated with their OpenML task ID.",
+    ),
+    filter_errors: bool = typer.Option(
+        False,
+        help="If True, any dataset missing a result from any framework in `frameworks` will be filtered from all results. Takes priority over `framework_nan_fill`.",
+    ),
+    clean_data: bool = typer.Option(
+        True,
+        help="If True, performs some general data cleanup to avoid inconsistencies in dataset naming compared to task ids.",
+    ),
+) -> Dict[str, pd.DataFrame]:
+    """
+    Generate evaluation results and saves the results to <results_dir>output/openml/<output_suffix>/evaluation_results.json
+
+    Example:
+    agbench evaluate-amlb-results --frameworks_run framework_1 --frameworks_run framework_2 --paths openml_ag_ag_bench_20230707T070230.csv --results-dir-input data/results/input/prepared/openml --no-clean-data
+    """
+    evaluate(
+        frameworks_run=frameworks_run if frameworks_run else None,
+        paths=paths if paths else None,
+        results_dir=results_dir,
+        results_dir_input=results_dir_input,
+        results_dir_output=results_dir_output,
+        output_suffix=output_suffix,
+        framework_nan_fill=framework_nan_fill,
+        problem_type=problem_types if problem_types else None,
+        folds_to_keep=folds_to_keep if folds_to_keep else None,
+        compute_z_score=compute_z_score,
+        treat_folds_as_datasets=treat_folds_as_datasets,
+        banned_datasets=banned_datasets if banned_datasets else None,
+        infer_batch_size=infer_batch_size,
+        clean_data=clean_data,
+        use_tid_as_dataset_name=use_tid_as_dataset_name,
+        filter_errors=filter_errors,
+    )
+
 
 # TODO: Rename to a more description function, or convert to a class
-def run(
+def evaluate(
     *,
     frameworks_run: List[str],
     paths: List[str],
+    results_dir: str = "data/results/",
+    results_dir_input: str = None,
+    results_dir_output: str = None,
     output_suffix: str = "ag_full_v5/1h8c",
-    framework_nan_fill: str | None = None,
-    problem_type: List[str] | str | None = None,
-    folds_to_keep: List[int] | None = None,
+    framework_nan_fill: Optional[str] = None,
+    problem_type: Union[List[str], str, None] = None,
+    folds_to_keep: Optional[List[int]] = None,
```

Comment:
`from __future__ import annotations` allows to use ` | ` for type hints to represent `or`, eliminating the need for `Optional[FOO]` , and instead using `FOO | None`, which is easier to read.

Reply:
Good to know! Will change back